federatedscope/gfl/trainer/graphtrainer.py

```diff
@@ -84,14 +84,14 @@ def _hook_on_batch_forward_flop_count(self, ctx):
         self.ctx.monitor.total_flops += self.ctx.monitor.flops_per_sample * \
             ctx.batch_size
 
-    def save_prediction(self, client_id, task_type):
+    def save_prediction(self, path, client_id, task_type):
         y_inds, y_probs = self.ctx.test_y_inds, self.ctx.test_y_prob
-        os.makedirs('prediction', exist_ok=True)
+        os.makedirs(path, exist_ok=True)
 
         # TODO: more feasible, for now we hard code it for cikmcup
         y_preds = np.argmax(y_probs, axis=-1) if 'classification' in task_type.lower() else y_probs
 
-        with open('prediction/prediction.csv', 'a') as file:
+        with open(path+'/prediction.csv', 'a') as file:
```

Comment:
please use `os.path.join` to concat strings into a valid path.

Reply:
Done